platform/ios/export/export_plugin.cpp

```diff
@@ -2260,6 +2421,82 @@ Error EditorExportPlatformIOS::_export_project_helper(const Ref<EditorExportPres
 		return ERR_FILE_NOT_FOUND;
 	}
 
+	// HACK: We don't want to run the simulator library generation code anymore after GH-102179, but removing it
+	// triggers an internal compiler error with latest mingw-gcc... For now we bring it back as a workaround
+	// with a condition that should never be true (that project setting doesn't exist).
+
+	// Check and generate missing ARM64 simulator library.
+	if (ProjectSettings::get_singleton()->has_setting("ios_generate_simulator_library_if_missing")) {
```

Comment:
This is the main difference with the previous code before #102179. Instead of checking a preset option (which #102179 removed), I'm checking a non-existing project setting. This should prevent the compiler from optimizing all this code away (it doesn't know that the project setting doesn't exist), but should also prevent it from running when exporting to iOS (unless a user decides to create this magic project setting manually).

This should really be a temporary hack (famous last words).